CMakeLists.txt

```diff
@@ -100,6 +100,19 @@ if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT AND CMAKE_PROJECT_NAME STREQUAL P
     endif(WIN32)
 endif()
 
+# Check to make sure the install prefix isn't the build folder, if it is, build errors will
+# happen
+get_filename_component(tmp_install_prefix ${CMAKE_INSTALL_PREFIX} REALPATH)
+get_filename_component(tmp_proj_bindir ${PROJECT_BINARY_DIR} REALPATH)
+# Windows paths are case insensitive
+if(WIN32)
+    string(TOLOWER "${tmp_install_prefix}" tmp_install_prefix)
+    string(TOLOWER "${tmp_proj_bindir}" tmp_proj_bindir)
+endif()
+if(tmp_install_prefix STREQUAL tmp_proj_bindir)
+    message(FATAL_ERROR "CMAKE_INSTALL_PREFIX must not be set to the build folder")
+endif()
+
```

Comment:
The `REALPATH` here is to resolve, symbolic links and relative paths in either variable?

Reply:
Yes, leaving a symlink or relative path in one but not the other could make it look like two paths are different when they are the same.

Hopefully this catches most of the cases. macOS is a special, since it can have case-sensitive or case-insensitive depending the filesystem the drive was formatted with; I didn't find a way to detect what the drive was formatted with in CMake.